specification/billing/resource-manager/Microsoft.Billing/stable/2020-05-01/billing.json

```diff
@@ -3460,6 +3460,150 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/createBillingRoleAssignment": {
+      "post": {
+        "tags": [
+          "BillingRoleAssignments"
+        ],
+        "operationId": "BillingRoleAssignments_AddByBillingAccount",
+        "description": "Adds a role assignment on a billing account. The operation is supported for billing accounts with agreement type Microsoft Customer Agreement.",
+        "x-ms-examples": {
+          "AddRoleAssignmentToBillingAccount": {
+            "$ref": "./examples/AddRoleAssignmentToBillingAccount.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/billingAccountNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BillingRoleAssignmentRequestProperties"
+            },
+            "description": "Request parameters that are provided to the create billing role assignment operation."
+          }
+        ],
+        "responses": {
+          "201": {
+            "description": "Role assignment is successfully created",
+            "schema": {
+              "$ref": "#/definitions/BillingRoleAssignment"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/billingProfiles/{billingProfileName}/invoiceSections/{invoiceSectionName}/createBillingRoleAssignment": {
```

Comment:
>createBillingRoleAssignment [](start = 145, length = 27)

Why are we modeling this as a POST instead of a proper resource with PUT and GET and list?

Reply:
The swagger already had the GET apis. We are currently adding an action to add the role assignments.
We have introduced this as POST since we rely on an external source to store the role assignments and they do not support PUT (do do not accept any user defined values as of now.) We tried some workarounds at our end to store it locally but it was not working with concurrent updates etc.

Reply:
That is the reason we did not publish this api along with other APIs

Reply:
Makes sense.